File: fast_mm_python/src/complete_split/EncodeCSD.py
"""
Encode an index array (i_1, ..., i_power) to a number in [1, 3^power].
"""

import numpy as np


def EncodeCSD(arr, power=None):
    """
    Encode an index array (i_1, ..., i_power) to a number in [1, 3^power].
    
    Args:
        arr: Index array, each element in {0, 1, 2}
        power: Power value (unused, for compatibility with MATLAB version)
    
    Returns:
        id: Encoded ID starting from 1
    """
    arr = np.atleast_1d(arr)
    id_val = 0
    for i in range(len(arr)):
        id_val = id_val * 3 + arr[i]
    return int(id_val) + 1  # Base-3 index starting from 1

File: fast_mm_matlab/src/complete_split/EncodeCSD.m
% Encode an index array (i_1, ..., i_{power}) to a number in [1, 3^power].

function id = EncodeCSD(arr, ~)
  id = 0;
  for i = 1 : length(arr)
    id = id * 3 + arr(i);
  end
  id = id + 1;  % base-3 index starting from 0
end
